command/validate.go

```diff
@@ -48,24 +54,39 @@ func (c *ValidateCommand) Synopsis() string {
 
 func (c *ValidateCommand) Help() string {
 	helpText := `
-Usage: terraform validate [options] [path]
+Usage: terraform validate [options] [dir]
+
+  Validate the terraform files in a directory. Validation includes a
+  basic check of syntax as well as checking that all variables declared
+  in the configuration are specified in one of the possible ways:
 
-  Reads the Terraform files in the given path (directory) and
-  validates their syntax and basic semantics.
+      -var foo=...
+      -var-file=foo.vars
+      TF_VAR_foo environment variable
+      terraform.tfvars
+      default value
 
-  This is not a full validation that is normally done with
-  a plan or apply operation, but can be used to verify the basic
-  syntax and usage of Terraform configurations is correct.
+  If dir is not specified, then the current directory will be used.
 
 Options:
 
+  -config-only        If specified, the command will check basic syntax of
```

Comment:
This flag name feels like it could hurt in the long run if `validate` were to get other tweakables like this.

Perhaps this could be `-check-variables=false`, with the default being `-check-variables=true`, and then if we add other things in the future they could be like `-check-types=false` or whatever, defaulting to everything being on by default.